libc::renameat2() gets invoked on musl platforms where libc-rs doesn't provide it

Build easy_fuser on Alpine Linux fails with "cannot find function `renameat2` in crate `libc`",

```
easy_fuser$ cargo build --features parallel
   Compiling easy_fuser v0.4.5 (/builder/repo/easy_fuser)
error[E0425]: cannot find function `renameat2` in crate `libc`
    --> src/unix_fs/linux_fs.rs:27:20
     |
  27 |       unsafe { libc::renameat2(olddirfd, oldpath, newdirfd, newpath, ...
     |                      ^^^^^^^^^
     |
    ::: /home/ziyao/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/libc-0.2.186/src/unix/mod.rs:1022:5
     |
1022 | /     pub fn renameat(
1023 | |         olddirfd: c_int,
1024 | |         oldpath: *const c_char,
1025 | |         newdirfd: c_int,
1026 | |         newpath: *const c_char,
1027 | |     ) -> c_int;
     | |_______________- similarly named function `renameat` defined here
     |
help: a function with a similar name exists
     |
  27 -     unsafe { libc::renameat2(olddirfd, oldpath, newdirfd, newpath, flags) }
  27 +     unsafe { libc::renameat(olddirfd, oldpath, newdirfd, newpath, flags) }
     |

For more information about this error, try `rustc --explain E0425`.
error: could not compile `easy_fuser` (lib) due to 1 previous error
easy_fuser$ rustc --version
rustc 1.95.0 (59807616e 2026-04-14) (Alpine Linux Rust 1.95.0-r0)
easy_fuser$ cat /etc/os-release 
NAME="Alpine Linux"
ID=alpine
VERSION_ID=3.24.0_alpha20260127
PRETTY_NAME="Alpine Linux edge"
HOME_URL="https://alpinelinux.org/"
BUG_REPORT_URL="https://gitlab.alpinelinux.org/alpine/aports/-/issues"
```

`renameat2()` API is only provided by libc-rs on Linux-GNU platforms (i.e., glibc), and redox, but not on musl-libc.

--- a/.github/workflows/ubuntu.yml
+++ b/.github/workflows/ubuntu.yml
@@ -72,3 +72,21 @@ jobs:
         PKG_CONFIG_ALLOW_CROSS: 1
         PKG_CONFIG_PATH: /usr/lib/i386-linux-gnu/pkgconfig
       run: cargo check --target=i686-unknown-linux-gnu --features parallel
+
+  check_musl:
+    if: github.event_name != 'pull_request' || github.event.pull_request.draft == false
+    name: "Check musl compilation"
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - name: Install musl-tools
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y musl-tools
+    - name: Setup Rust
+      run: |
+        rustup update stable
+        rustup default stable
+        rustup target add x86_64-unknown-linux-musl
+    - name: Check
+      run: cargo check --target=x86_64-unknown-linux-musl --features parallel

--- a/src/unix_fs/linux_fs.rs
+++ b/src/unix_fs/linux_fs.rs
@@ -23,7 +23,21 @@ pub(super) unsafe fn renameat2(
     newpath: *const c_char,
     flags: c_uint,
 ) -> c_int {
-    unsafe { libc::renameat2(olddirfd, oldpath, newdirfd, newpath, flags) }
+    #[cfg(target_env = "gnu")]
+    unsafe {
+        libc::renameat2(olddirfd, oldpath, newdirfd, newpath, flags)
+    }
+    #[cfg(not(target_env = "gnu"))]
+    unsafe {
+        libc::syscall(
+            libc::SYS_renameat2,
+            olddirfd,
+            oldpath,
+            newdirfd,
+            newpath,
+            flags,
+        ) as c_int
+    }
 }
 
 pub(super) unsafe fn fdatasync(fd: c_int) -> c_int {

